Revised game interactions

joelwross opened the pull request:
Also merges with other work done in this branch.

Note that the logging is currently hard-coded to a specific directory, change it in game_controller.rb to get it to work on your local machine (please fix this Duyet?)

joelwross committed:
finished restoring changes overwritten by faulty merge I think.

joelwross committed:
ants change color when dead or settled; settled ants move around with islands

joelwross committed:
output correctness to javascript (my god); began implementing islands

joelwross committed:
islands now deploy ants along their (still pre-determined) paths, rather than having the ants make their own decisions. Also some animation refactoring

joelwross committed:
islands deploy ants with delay

joelwross committed:
new ant movement behavior (go to island then settle) finished

joelwross committed:
ants now wander off the islands if there are no bridges for them to take. Also debugged some stuff, so we have something that strongly resembles continuous play

joelwross committed:
bug fixes, and added a sensible scoreboard method. Still a few issues to tweak, but this form of the game is close to done. This commit will be pushed to branch for others to poke at

joelwross committed:
resolved merge conflicts

joelwross committed:
continuing to resolve merge conflicts

joelwross committed:
resolve merge conflicts

joelwross committed:
Merge branch 'master' of git@github.com:rgoraya/wc_master

joelwross committed:
added introduction to game

joelwross committed:
some cleanup to game text displays. MAJOR refactor of game javascript code (so it even less tied to rails, and doesn't use ajax where not needed)

joelwross committed:
added continuous mode commands, fixed bridge cycling

dLobatog commented:
Can you check out the game_controller.rb that is now on master and change the few lines relative to the absolute path of the logging and the environment and update the pull request?

joelwross committed:
changed pacing behavior to circling, added some initial times to continuous mode (seems fast though; will have to figure out the right times, probably via in-house testing?

joelwross commented:
The version I have from my branch (where the rest of us are working on the game code) has updated relative path. I imagine that is the stuff you're looking for?

joelwross committed:
Merge branch 'master' of git@github.com:rgoraya/wc_master

Commit:
Game welcome screens improved--CSS Buttons, styling, etc

-new all-css buttons added
-positioning improved

known bug:
can't seem to get the entire window to have a background of #9dcec5,
something to do with how HAML generates its divs and how I can't figure
out where to point the css

joelwross committed:
Merge pull request #30 from theSahand/master

Welcome page improvements

Commit:
Quick CSS Fix

removed the 1 & 2 on the left side of the game's welcome screen CSS
buttons, per request

also removed paragraph double-spacing

joelwross committed:
Merge pull request #31 from theSahand/master

Quick CSS Fix

joelwross committed:
layout text tweaks to introduction

joelwross committed:
minor cleanup on dragging edges (icons now appear at correct depth; couldn't get new edges to show up on top though). Added 'hesitate' command to the ants--they check validity at every step now, and if path is invalid wait a moment before plunging to a watery death

Commit:
Added sounds, in order to avoid concurrency problems they only play around 10% of the times

duyetln committed:
Island box is modified

joelwross committed:
Merge pull request #33 from nguyendl/master

Island box is modified

dLobatog committed:
Merge branch 'master' of git://github.com/highbulp/wc_master into audio-in-game

Conflicts:
	app/views/game/run.js.erb
	webrat.log

dLobatog committed:
Removed run.js and added the audio control to game.js

dLobatog committed:
Only running one simulation at a time is allowed

joelwross committed:
Merge pull request #34 from eLobato/master

Sounds code and validation before running simulation

joelwross committed:
new design for islands

joelwross committed:
began adding hooks to including research surveys within the game

joelwross committed:
we now validate indirect edges as correct; level of indirection can be changed in the Game model

dLobatog committed:
Changes for the bar buttons and background

dLobatog committed:
Merge branch 'master' of git://github.com/highbulp/wc_master

duyetln committed:
Improved interactions with selection box; fixed a bug where islands are not translated properly; added arrows and wooden background

duyetln committed:
Some tweaks to arrows transformation

joelwross committed:
Merge pull request #36 from nguyendl/master

Improved interactions with selection box; wood background; arrows

joelwross committed:
why is game_log not being ignored?

joelwross committed:
Merge branch 'master' of git@github.com:highbulp/wc_master

dLobatog committed:
necessary to merge

dLobatog committed:
necessary to merge

dLobatog committed:
Adapted to Sahand's design. Islands now are dragged from the bottom bar

joelwross committed:
Merge pull request #37 from eLobato/master

Adapted to Sahand's design

joelwross committed:
trying to fix game_log merge

joelwross committed:
tryingto fix log again

joelwross committed:
added help.png back in after merge problems

dLobatog committed:
Buttons shine on hover, help is loaded by default

joelwross committed:
Merge pull request #38 from eLobato/master

Email changes

joelwross committed:
Merge branch 'master' of git@github.com:highbulp/wc_master

joelwross committed:
redesigned edges; added a hook for 'how to play' page

joelwross committed:
added full text display to islands when they are on the dock

joelwross committed:
slight resizing to make it fit on the screen better...

joelwross committed:
double-click to swap directions
